### PR TITLE
docs(profiling): Add page about performance overhead

### PR DESCRIPTION
**Pull request by indragiek**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

Add a page describing the performance overhead of enabling profiling in production.

### CODE REVIEW

File: src/docs/product/profiling/performance-overhead.mdx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+title: Performance Overhead
+sidebar_order: 4
+redirect_from:
+  - /profiling/performance-overhead/
+description: "How enabling profiling impacts the performance of your application."
+---
+
+Profiling works by taking snapshots (samples) of the call stack of each thread in your application at a regular interval, and transmitting that data to event ingestion servers over the public internet. This method of profiling is known as sampling profiling, and all of our profilers collect samples approximately 100 times per second (or once every 10ms). Sampling profiling provides a reasonable tradeoff between data fidelity and runtime overhead, as opposed to deterministic profiling, which introduces significantly higher overhead by tracking every function call.
+
```
Comment (lizokm) [inline src/docs/product/profiling/performance-overhead.mdx:10]:
```suggestion
Profiling works by taking snapshots (samples) of the call stack of each thread in your application at a regular interval, then transmitting that data to event ingestion servers via the Internet. This method of profiling is known as sampling profiling. Sentry's profilers collect samples approximately 100 times per second (or once every 10ms). Sampling profiling provides a reasonable tradeoff between data fidelity and runtime overhead, as opposed to deterministic profiling, which introduces significantly higher overhead by tracking every function call.

```

File: src/docs/product/profiling/performance-overhead.mdx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+**For most applications, the performance overhead of enabling profiling in production is imperceptible to end users.**
+
```
Comment (lizokm) [inline src/docs/product/profiling/performance-overhead.mdx:12]:
```suggestion
<Note>
For most applications, the performance overhead of enabling profiling in production is imperceptible to end users.
</Note>

```

File: src/docs/product/profiling/performance-overhead.mdx [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+The quantitative overhead of profiling (measured as a % increase in CPU time) is highly dependent on many environmental factors such as the SDK, the platform, the hardware, and whether the workload is CPU or I/O bound. However, as a general rule of thumb, we target an approximate **5-10% CPU time overhead** on a typical configuration with mid-end hardware on each platform.
```
Comment (lizokm) [inline src/docs/product/profiling/performance-overhead.mdx]:
```suggestion
The quantitative overhead of profiling, (measured as a % increase in CPU time), is highly dependent on many environmental factors such as: the SDK, the platform, the hardware, and whether the workload is CPU or I/O bound. However, as a general rule of thumb, Sentry targets an approximate **5-10% CPU time overhead** on a typical configuration with mid-end hardware on each platform.
```

**Review by lizokm — APPROVED** [review]
A few small tweaks, but otherwise this looks good!

File: src/docs/product/profiling/performance-overhead.mdx [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+redirect_from:
+  - /profiling/performance-overhead/
+description: "How enabling profiling impacts the performance of your application."
+---
+
+Profiling works by taking snapshots (samples) of the call stack of each thread in your application at a regular interval, then transmitting that data to event ingestion servers via the Internet. This method of profiling is known as sampling profiling. Sentry's profilers collect samples approximately 100 times per second (or once every 10ms). Sampling profiling provides a reasonable tradeoff between data fidelity and runtime overhead, as opposed to deterministic profiling, which introduces significantly higher overhead by tracking every function call.
+
+<Note>
+For most applications, the performance overhead of enabling profiling in production is imperceptible to end users.
+</Note>
```
Comment (shanamatthews) [inline src/docs/product/profiling/performance-overhead.mdx:13]:
```suggestion
<Note>

For most applications, the performance overhead of enabling profiling in production is imperceptible to end users.

</Note>
```

Comment (shanamatthews) [reply src/docs/product/profiling/performance-overhead.mdx:13]: I remember reading that we need these extra newlines for weird Gatsby reasons

File: src/docs/product/profiling/performance-overhead.mdx [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+The quantitative overhead of profiling (measured as a % increase in CPU time) is highly dependent on many environmental factors such as the SDK, the platform, the hardware, and whether the workload is CPU or I/O bound. However, as a general rule of thumb, we target an approximate **1-5% CPU time overhead** on a typical configuration with mid-end hardware on each platform.
```
Comment (shanamatthews) [inline src/docs/product/profiling/performance-overhead.mdx:15]:
```suggestion
The quantitative overhead of profiling (measured as a percentage increase in CPU time) is dependent on environmental factors such as the SDK, the platform, the hardware, and whether the workload is CPU or I/O bound. As a rule of thumb, we target approximately **1-5% CPU time overhead** on a typical configuration with mid-end hardware on each platform.
```

Comment (shanamatthews) [reply src/docs/product/profiling/performance-overhead.mdx:15]: this felt wordy, so I simplified a bit

**Review by shanamatthews — APPROVED** [review]

**Pull request merged**